modin/core/dataframe/algebra/binary.py

```diff
@@ -36,6 +120,13 @@ def register(cls, func, join_type="outer", labels="replace"):
         labels : {"keep", "replace", "drop"}, default: "replace"
             Whether keep labels from left Modin DataFrame, replace them with labels
             from joined DataFrame or drop altogether to make them be computed lazily later.
+        how_compute_dtypes : {"common_cast", "float","bool", None}, default: None
+            How dtypes should be computed.
+            If "common_cast" casts to common dtype of operand columns.
+            If "float" performs type casting by finding common dtype, if the common dtype is any of the
+            integer types perform type casting to float. Used in case of truediv.
+            If "bool" dtypes would be a boolean series with same size as that of operands.
+            If ``None`` do not infer new dtypes (they will be computed manually once accessed).
```

Comment:
```suggestion
        infer_dtypes : {"common_cast", "float","bool", None}, default: None
            How dtypes should be inferred.
                * If "common_cast", casts to common dtype of operand columns.
                * If "float", performs type casting by finding common dtype.
                   If the common dtype is any of the integer types, perform type casting to float.
                   Used in case of truediv.
                * If "bool", dtypes would be a boolean series with same size as that of operands.
                * If ``None``, do not infer new dtypes (they will be computed manually once accessed).
```